### ISSUE TITLE
podman search fails after login

### ISSUE THREAD

**Issue by thomasmckay** [projectatomic/libpod#976]
Note that the failed search behavior below is against a newly-added feature to foreman so it could be that there is missing headers or other response information that podman is expecting. I don't have another auth-backed registry to test against to compare, though.

```
$ podman version
Version:       0.4.1
Go Version:    go1.9.4
OS/Arch:       linux/amd64

$ sudo podman login -u developer -p redhat devel.example.com 
Login Succeeded!

$ sudo podman search --registry devel.example.com /
ERRO[0000] error searching registry "devel.example.com": couldn't search registry "devel.example.com": unable to retrieve auth token: invalid username/password 
```
_podman pull_ worked fine after the login.

From registry server logs
```
2018-06-20T23:02:18 [I|app|] Started GET "/v2/" for 127.0.0.1 at 2018-06-20 23:02:18 +0000
2018-06-20T23:02:18 [I|app|] Started GET "/v2/token?service=devel.example.com" for 127.0.0.1 at 2018-06-20 23:02:18 +0000
2018-06-20T23:02:18 [I|app|] Started GET "/v2/token?service=devel.example.com" for 127.0.0.1 at 2018-06-20 23:02:18 +0000
```

**Comment by rhatdan** [comment]
@haircommander PTAL

**Comment by umohnani8** [comment]
Fixed in #998
